homeassistant/components/abode/__init__.py

```diff
@@ -131,17 +129,50 @@ def is_light(self, device):
         )
 
 
-def setup(hass, config):
+async def async_setup(hass, config):
     """Set up Abode component."""
-    from abodepy.exceptions import AbodeException
+    if DOMAIN not in config:
+        return True
+
+    hass.data[DOMAIN] = {}
 
     conf = config[DOMAIN]
-    username = conf.get(CONF_USERNAME)
-    password = conf.get(CONF_PASSWORD)
-    name = conf.get(CONF_NAME)
-    polling = conf.get(CONF_POLLING)
-    exclude = conf.get(CONF_EXCLUDE)
-    lights = conf.get(CONF_LIGHTS)
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_USERNAME: conf.get(CONF_USERNAME),
+                CONF_PASSWORD: conf.get(CONF_PASSWORD),
+                CONF_NAME: conf.get(CONF_NAME),
+                CONF_POLLING: conf.get(CONF_POLLING),
+                CONF_EXCLUDE: conf.get(CONF_EXCLUDE),
```

Comment:
We don't allow excluding specific devices/entities in the config anymore. We can instead use entity disabled options. If possible we should set a default way and include all entities according to that default.
https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md
https://developers.home-assistant.io/docs/en/entity_registry_disabled_by.html

Reply:
By default, it appears all entities are included which I think is fine since users can then disable entities from the UI (entities page). Would this suffice just relying on the defaults for now? Ideally, I think it would be nice to allow users to select which devices they want to disable during the config entry process, but I'm not quite sure how to do that. The options config flow seems to just be an options dialog for the entire component. I'm not sure if it's possible to have configurable options for each individual device within an integration?

Reply:
Yes, I think it's enough in this PR to remove these options and let the user disable entitites if needed afterwards.

We don't want options for each configurable device in the options flow. If we can group devices somehow to disable that could be an option. Individual entities are disabled via the options in entity registry.

Reply:
After removing code related to exclude and lights, everything works fine but one unexpected outcome is all switches from Abode show up as one device in HA but two entities, one switch and one light (even if it's not necessarily a light). A user can simply disable the one they don't need, but I'd like to better understand why this is occurring. Will keep looking into it.

As a side note, this will be a "major" change for current users of the Abode component as I'm sure many users have excluded devices in their config file.